spec/models/activity_spec.rb

```diff
@@ -2266,26 +2266,37 @@ def create_report(organisation, fund)
     end
 
     context "when the activity is an ODA ISPF programme" do
-      it "returns non-ODA ISPF programmes with the same PO as their extending organisation" do
+      it "returns unlinked non-ODA ISPF programmes with the same PO as their extending organisation" do
         oda_programme = create(:programme_activity, :ispf_funded, extending_organisation: partner_organisation, is_oda: true)
         non_oda_programme = create(:programme_activity, :ispf_funded, extending_organisation: partner_organisation, is_oda: false)
         _other_po_non_oda_programme = create(:programme_activity, :ispf_funded, is_oda: false)
         _other_oda_programme = create(:programme_activity, :ispf_funded, is_oda: true)
+        _non_oda_programme_linked = create(:programme_activity, :ispf_funded, extending_organisation: partner_organisation, is_oda: false, linked_activity: build(:programme_activity))
```

Comment:
I was trying to keep the test minimalist, but you're right 😅 Tests should reflect a realistic setup. And if we ever add model-level validation for linking, tests like this would suddenly fail...